### PR TITLE
feat: polling profiles endpoint

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR adds the polling profiles resource endpoint to handle the CRUD operations

Part of #2009 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: server/resourcemanager/resource_manager.go [hunk 1]
```diff
@@ -110,18 +110,21 @@ func (m *manager[T]) RegisterRoutes(r *mux.Router) *mux.Router {
 	enabledOps := m.EnabledOperations()
 
 	if slices.Contains(enabledOps, OperationList) {
-		subrouter.HandleFunc("/", m.list).Methods(http.MethodGet).Name("list")
+		subrouter.HandleFunc("", m.list).Methods(http.MethodGet).Name("list")
 	}
+
 	if slices.Contains(enabledOps, OperationCreate) {
-		subrouter.HandleFunc("/", m.create).Methods(http.MethodPost).Name(fmt.Sprintf("%s.Create", m.resourceType))
```
Comment (xoscar) [inline server/resourcemanager/resource_manager.go:116]: The obscured hidden bug haha great catch!

**Review by xoscar — APPROVED** [review]

File: server/resourcemanager/resource_manager.go (hunk 1 above)
Comment (mathnogueira) [reply server/resourcemanager/resource_manager.go:116]: As @danielbdias likes to say: This like of code owns me 3 hours of debugging

File: server/app/app.go [hunk 2]
```diff
@@ -283,6 +287,15 @@ func registerConfigResource(configRepo *configresource.Repository, router *mux.R
...
 }
 
+func registerPollingProfilesResource(configRepo *pollingprofile.Repository, router *mux.Router, db *sql.DB) {
+	manager := resourcemanager.New[pollingprofile.PollingProfile](
+		pollingprofile.ResourceName,
+		configRepo,
+		resourcemanager.WithOperations(pollingprofile.Operations...),
+	)
+	manager.RegisterRoutes(router)
+}
```
Comment (danielbdias) [inline server/app/app.go:297]: We can rename `configRepo` here to just `repo`:

```suggestion
func registerPollingProfilesResource(repo *pollingprofile.Repository, router *mux.Router, db *sql.DB) {
	manager := resourcemanager.New[pollingprofile.PollingProfile](
		pollingprofile.ResourceName,
		repo,
		resourcemanager.WithOperations(pollingprofile.Operations...),
	)
	manager.RegisterRoutes(router)
}
```

File: server/executor/pollingprofile/polling_profile_resource.go [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+func (r *Repository) Create(ctx context.Context, profile PollingProfile) (PollingProfile, error) {
+	var (
+		periodicJSON []byte
+		err          error
+	)
+
+	if profile.ID == "" {
+		profile.ID = id.SlugFromString(profile.Name)
+	}
```
Comment (danielbdias) [inline server/executor/pollingprofile/polling_profile_resource.go:85]: (to think in future) Later we need to map this behavior to document it somewhere. (maybe on OpenAPI specs?)

File: server/executor/pollingprofile/polling_profile_resource.go [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+	if profile.Default {
+		err := r.clearDefaultFlag(ctx)
+		if err != nil {
+			return PollingProfile{}, fmt.Errorf("could not update old default profile: %w", err)
+		}
+	}
+
+	if profile.Periodic != nil {
```
Comment (danielbdias) [inline server/executor/pollingprofile/polling_profile_resource.go:94]: Should we validate the presence of this field and throw an error? Or will we assume a default behavior if the strategy is not sent?

**Review by danielbdias — APPROVED** [review]

**Pull request merged**